docs/design/maven.md

```diff
@@ -0,0 +1,130 @@
+# Generic artifact fetching
+
+## Introduction
+
+This document will describe high-level implementation overview for supporting fetching maven artifacts in cachi2. This
+proposal depends on generic artifact fetching, which is described in this [design PR][generic-pr].
+
+## Context
+
+[Maven] is a Java ecosystem build system and functions basically as a package manager, among other things. Maven provides a
+standardized way to specify dependencies in a `pom.xml` file, which is then used to fetch dependencies from a maven repository.
+This feature in cachi2 isn't concerned with pre-fetching those, but rather to provide a way to fetch individual artifacts
+from maven repository. This is useful for cases where the build only needs specific artifacts from big projects, saving
+time and bandwidth.
+
+### Maven artifacts
+
+[Maven artifacts][maven-artifacts] are defined as anything that resides in a maven repository. Maven has a special way of
+addressing artifacts in its repositories called coordinates, which is usually represented by `GAV` (Group, Artifact, Version)
+This is used to uniquely identify an artifact in a maven repository.
+
+### Maven purls
+
+Purls for `pkg:maven` ([docs][maven-purl]) fully describe the location of an artifact in a maven repository. That is because
+all parts of the purl (`namespace`, `name`, `version` and the `classifier` and `type` qualifiers) are used to construct
+the download url for the artifact. [Maven documentation][maven-repo-layout] describes how to turn these coordinates into
```

Comment:
This is a bit confusing. Should purls be translated to GAVs or the other way around?

Reply:
I suggest adding this explanation to this doc, this is important for understanding the specifics of the proposed package manager.

Reply:
Added.